### PR TITLE
Feature/zapier integration

### PR DESCRIPTION
**Pull request by acoumb**
More information on the setup and integration details have been sent by email.

### CODE REVIEW

**Review by AndyButland — COMMENTED** [review]
Code looks good, have placed a few points inline.

Currently trying to test it and am running into an issue though - this is what I see in the logs:

```
System.FormatException: Unrecognized Guid format.
   at System.Guid.GuidResult.SetFailure(ParseFailureKind failure, String failureMessageID, Object failureMessageFormatArgument, String failureArgumentName, Exception innerException)
   at System.Guid.TryParseGuid(String g, GuidStyles flags, GuidResult& result)
   at System.Guid.Parse(String input)
   at Umbraco.Forms.Integrations.Automation.Zapier.Services.MappingService.IncludeFieldsMappings(String mappings, RecordEventArgs e)
   at Umbraco.Forms.Integrations.Automation.Zapier.ZapierWorkflow.Execute(Record record, RecordEventArgs e)
```

File: src/Umbraco.Forms.Integrations.Automation.Zapier/ZapierWorkflow.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            try
+            {
+                IMappingService mappingService = new MappingService(_umbracoContextAccessor);
+
+                var content = mappingService
+                    .IncludeFieldsMappings(Mappings, e)
+                    .IncludeStandardFieldsMappings(StandardFieldsMappings, e)
+                    .Map();
+
+                var result = ClientFactory().PostAsync(WebHookUri, new FormUrlEncodedContent(content)).Result;
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Forms.Integrations.Automation.Zapier/ZapierWorkflow.cs]: Updated

File: src/Umbraco.Forms.Integrations.Automation.Zapier/Controllers/AuthController.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        public bool ValidateUser([FromBody] UserModel userModel)
+        {
+            var isUserValid = Security.ValidateBackOfficeCredentials(userModel.Username, userModel.Password);
+            if (!isUserValid) return false;
+
+            IUserService userService = Services.UserService;
+
+            var user = userService.GetByUsername(userModel.Username);
+
+            return user != null && user.Groups.Any(p => p.Alias == "admin");
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Forms.Integrations.Automation.Zapier/Controllers/AuthController.cs]: Amended the Zapier app and addressed the logic.

File: src/Umbraco.Forms.Integrations.Automation.Zapier/Umbraco.Forms.Integrations.Automation.Zapier.csproj [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		<RepositoryUrl>https://github.com/umbraco/Umbraco.Forms.Integrations</RepositoryUrl>
+		<Version>1.0.0</Version>
+		<Authors>Umbraco HQ</Authors>
+		<Company>Umbraco</Company>
+	</PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="UmbracoCms.Web" Version="8.1.0" />
+	<PackageReference Include="UmbracoCms.Core" Version="8.1.0" />
+    <PackageReference Include="UmbracoForms.Core" Version="8.11.0" />
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Forms.Integrations.Automation.Zapier/Umbraco.Forms.Integrations.Automation.Zapier.csproj]: As I am using StandardFieldMapping references which were included in version 8.9.1, I have lowered the dependency to this value.

File: src/Umbraco.Forms.Integrations.Automation.Zapier/Services/MappingService.cs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+using Newtonsoft.Json;
+
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Forms.Core;
+using Umbraco.Forms.Core.Providers.Models;
+using Umbraco.Web;
+
+namespace Umbraco.Forms.Integrations.Automation.Zapier.Services
+{
+    public class MappingService: IMappingService
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Forms.Integrations.Automation.Zapier/Services/MappingService.cs]: Updated. And also included the option to include all form fields if no mapping is added.

File: src/Umbraco.Forms.Integrations.Automation.Zapier/ZapierWorkflow.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        private readonly static HttpClient s_client = new HttpClient();
+
+        // Access to the client within the class is via ClientFactory(), allowing us to mock the responses in tests.
+        public static Func<HttpClient> ClientFactory = () => s_client;
+
+        public override WorkflowExecutionStatus Execute(Record record, RecordEventArgs e)
+        {
+            try
+            {
+                IMappingService mappingService = new MappingService(_umbracoContextAccessor);
```
[the remark this answers is not in the document]
Comment (acoumb) [reply src/Umbraco.Forms.Integrations.Automation.Zapier/ZapierWorkflow.cs]: Corrected.

**Comment by acoumb** [comment]
This exception _System.FormatException: Unrecognized Guid format._ has been fixed with a patch allowing static mappings to be included to the request dictionary.

**Comment by acoumb** [comment]
Moved user group to a configuration value. V9 will be managed on a different branch.

**Pull request merged**